modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java

```diff
@@ -336,19 +347,16 @@ public void setWorkflowService(WorkflowService workflowService) {
     this.workflowService = workflowService;
   }
 
-
   private List<EventCatalogUIAdapter> getEventCatalogUIAdapters() {
-    return new ArrayList<>(getEventCatalogUIAdapters(getSecurityService().getOrganization().getId()));
+    return getEventCatalogUIAdapters(getSecurityService().getOrganization().getId());
   }
 
-  public List<EventCatalogUIAdapter> getEventCatalogUIAdapters(String organization) {
-    List<EventCatalogUIAdapter> adapters = new ArrayList<>();
-    for (EventCatalogUIAdapter adapter : catalogUIAdapters) {
-      if (adapter.handlesOrganization(organization)) {
-        adapters.add(adapter);
-      }
-    }
-    return adapters;
+  private List<EventCatalogUIAdapter> getEventCatalogUIAdapters(String organization) {
+    List<EventCatalogUIAdapter> cachedCatalogUIAdapters = orgCatalogUIAdaptersMap.computeIfAbsent(organization,
+        org -> new ArrayList<>(catalogUIAdapters.stream()
+            .filter(a -> a.handlesOrganization(org))
+            .collect(Collectors.toList())));
+    return new CopyOnWriteArrayList<>(cachedCatalogUIAdapters);
```

Comment:
Why return a CopyOnWriteArrayList? Is the Instance used with multithreading?

Reply:
The list might be changed (some callers remove stuff). This should not affect other clients.

Reply:
But are those callers from different threads? Maybe I have overseen something, but each thread has its list and does not need a thread-safe version.